_posts/2021-01-05-git-rebase-merge.md

```diff
@@ -0,0 +1,167 @@
+
+## Introduction to Git
+
+Git is an **open-source distributed version control system**. We can break that work down into the following pieces:
+
+* **Control System:** Git can be used to store content – it is usually used to store code, but other content can also be stored.
+* **Version Control System:** Git helps in maintaining a history of changes and supports working on the same files in parallel by providing features like branching and merging.
+* **Distributed Version Control System:** The code is present in two types of repositories – the local repository, and the remote repository.
+
+## What is Git Merge?
+
+Now it's time to have a look at Git `merge`. A `merge` is a way to put a forked history back together. The git `merge` command lets us take independent branches of development and combine them into a single branch.
+
+It's important to note that while using the Git `merge`, the current branch will be updated to reflect the merge, but the target branch remains untouched. 
+
+`git merge` is often used in combination with `git checkout` for the selection of the current branch, and `git branch -d` for deleting the obsolete target branch.
+
+`git merge` is used for combining the multiple sequences of commits into one unified history. In the most common cases, we use `git merge` to combine two branches. 
+
+Let's take an example in which we will mainly focus on branch merging patterns. In the scenario which we have taken, `git merge` takes two commit pointers, and tries to find the common base commit between them. 
+
+Once Git has found a common base commit, it will create a new "merge commit", that will combine the changes of each queued merge commit sequence.
+
+![Git Merge combining branches](/assets/img/posts/git-rebase-merge/git-merge-working1.png)
+
+![Git Merge with Common Base](/assets/img/posts/git-rebase-merge/git-merge-working2.png)
+
+
+## What is Git Rebase?
+
+Let's have a look at the concept of Git `rebase`. A `rebase` is the way of migrating or combining a sequence of commits to a new base commit. If we consider it in the context of feature branching workflow, then it is more useful and easily visualized. The figure given below describes the general process of rebasing with Git:
+
+![Git Rebasing](/assets/img/posts/git-rebase-merge/git-rebasing-basic.png)
+
+Let's understand the working of Git `rebase` by looking a history with a topic branch off another topic branch. For example, we have branched a topic branch (feature2) from the mainline , and added some feature2-side functionality to our project, and then made a commit. Now, we branch off the `feature2` branch to make some client-side changes. Finally, we go back to the feature2-side branch and commit a few more changes:
```

Comment:
It looks like you missed some spots after search&replacing "client" and "server".  The text still contains "client" and says that we're branching `feature2` off master, when the image says that we're branching `feature` off master and `feature2` off `feature`. Please check that the text matches the image

Reply:
Also, the time aspect explained in the text (i.e. first we commit to one branch, then to another) is not visible in the image.

Reply:
I have updated the feature1 and feature2 branch details and now they are correct, previously I had made a mistake regarding the replacement and that's why nothing made sense, I hope this will be pretty clear now